### PR TITLE
feat: integrate halo

### LINKED ISSUE #388: Integrate Halo Module

### PR DESCRIPTION
**Pull request by boojamya**
Closes #388

### CODE REVIEW

File: app/upgrades/xenon/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		vm, err := mm.RunMigrations(ctx, cfg, vm)
+		if err != nil {
+			return vm, err
+		}
+
+		switch ctx.ChainID() {
+		case TestnetChainID:
+			halokeeper.SetOwner(ctx, "noble1u0nahk4wltsp89tpce4cyayd63a69dhpkfq9wq")
+			florinkeeper.SetOwner(ctx, "noble1tv9u97jln0k3anpzhahkeahh66u74dug302pyn")
```
Comment (johnletey) [inline app/upgrades/xenon/upgrade.go:73]: We need to set the owners of the submodules as well!

```suggestion
			halokeeper.SetOwner(ctx, "noble1u0nahk4wltsp89tpce4cyayd63a69dhpkfq9wq")
			halokeeper.SetAggregatorOwner(ctx, "noble1u0nahk4wltsp89tpce4cyayd63a69dhpkfq9wq")
			halokeeper.SetEntitlementsOwner(ctx, "noble1u0nahk4wltsp89tpce4cyayd63a69dhpkfq9wq")

			florinkeeper.SetOwner(ctx, "noble1tv9u97jln0k3anpzhahkeahh66u74dug302pyn")
			florinkeeper.SetBlacklistOwner(ctx, "noble1tv9u97jln0k3anpzhahkeahh66u74dug302pyn")
```

File: app/upgrades/xenon/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		Name:    "",
+		Symbol:  "USYC",
+	}
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	cfg module.Configurator,
+	halokeeper *halokeeper.Keeper,
+	florinkeeper *florinkeeper.Keeper,
```
Comment (johnletey) [inline app/upgrades/xenon/upgrade.go:61]: Let's make sure to not override the imports here ...

```suggestion
	haloKeeper *halokeeper.Keeper,
	florinKeeper *florinkeeper.Keeper,
```

File: app/upgrades/xenon/upgrade.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	florinkeeper "github.com/noble-assets/florin/x/florin/keeper"
+	halokeeper "github.com/noble-assets/halo/x/halo/keeper"
+)
+
+var (
+	// TODO: Verify denom metadata
+
+	eureMetaData = banktypes.Metadata{
```
Comment (johnletey) [inline app/upgrades/xenon/upgrade.go:18]: Just a style choice here

```suggestion
	eureMetadata = banktypes.Metadata{
```

File: app/upgrades/xenon/upgrade.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+				Exponent: 18,
+			},
+		},
+		Base:    "aeure",
+		Display: "eure",
+		Name:    "Euro Stablecoin",
+		Symbol:  "EURe",
+	}
+
+	usycMetaData = banktypes.Metadata{
```
Comment (johnletey) [inline app/upgrades/xenon/upgrade.go:37]: Ditto ...

```suggestion
	usycMetadata = banktypes.Metadata{
```

File: .github/workflows/e2e-tests.yaml [hunk 5]
```diff
@@ -15,13 +15,14 @@ jobs:
...
       - name: Build Docker Image
-        uses: strangelove-ventures/heighliner-build-action@v1.0.2
+        uses: strangelove-ventures/heighliner-build-action@v1.0.3
         with:
           registry: "" # empty registry, image only shared for e2e testing
           tag: local # emulate local environment for consistency in interchaintest cases
           tar-export-path: ${{ env.TAR_PATH }} # export a tarball that can be uploaded as an artifact for the e2e jobs
           platform: linux/amd64 # test runner architecture only
           git-ref: ${{ github.head_ref }} # source code ref
+          clone-key: ${{ secrets.AURA_SSH_KEY }}
```
Comment (johnletey) [inline .github/workflows/e2e-tests.yaml:25]: Could we rename this, as we're not dealing with Aura in this case.

**Review by johnletey — COMMENTED** [review]
Great job @boojamya, left a few comments!

File: app/upgrades/xenon/upgrade.go (hunk 2 above)
Comment (boojamya) [reply app/upgrades/xenon/upgrade.go]: Good eye

**Review by g-luca — APPROVED** [review]

**Pull request merged**